### PR TITLE
Fixes to make it possible to reproduce the paper

### PR DESCRIPTION
**Pull request by szobov**
Fixed parts:
* Graph sampling:
  - Use obj_io to avoid bugs with OBJ reading
  - Don't force people to mark vertices with colors, just mark some in code
  - Check for zero division in case of output points size is zero
* GraphAutoEncoder
  - remove whitespaces
  - make variables names more meaningful
  - remove unused parameters
  - change the logic of template mesh, now we don't require such logic, because we calculate normals for every triangle
  - remove logic of hardcoded list of training objects, instead, we specify the path to directory with .OBJ files
  - change the logic of calculating face normals to calculate normals per sample
  - add logic to draw histograms and plots for the gradients to understand gradients behaviour
  - output meshes to tensorboard to understand what network produces
  - add more output to tensorboard
* Add INSTRUCTION.md

Unfortunately, even after all these fixes, I wasn't able to reproduce this paper.
I had only one successful setup when I trained on the set of **1** mesh with batch size 2, in this case on the 3500 steps I had such reconstructed mesh, which is pretty nice:

image

In all other setups on the bigger dataset, the network didn't converge and was just stuck at some point, producing such junk:

image

When I tried to debug such behaviour, I found that the gradients for the encoder are vanishing and were fluctuating around 
zero:

image

While decoder still had some gradients different from zero:

image

Even when I left it learning for longer time (300k iterations with butch size 8) the result was the same:

image

I also tried different weights for loses and different structure of the network (using different numbers of layers) but has no success.

So, I decided to make a merge request with my changes to give other chances to continue fixin' this repo.

If authors could make any comments on my changes or give me insights or comments, I would highly appreciate it!

Cheers,
Sergei

### CODE REVIEW

**Comment by EJShim** [comment]
Is "Don't force people to mark vertices with colors, just mark some in code" means that you randomly mark some colored vertices from the input OBJ? If so, couldn't this method be the reason that this model cannot converged? I did not fully understood this method, but I feel the marked vertices should be something meaningful feature in lower-dimensional space of pooled layer.

**Comment by szobov** [comment]
> Is "Don't force people to mark vertices with colors, just mark some in code" means that you randomly mark some colored vertices from the input OBJ? If so, couldn't this method be the reason that this model cannot converged? I did not fully understood this method, but I feel the marked vertices should be something meaningful feature in lower-dimensional space of pooled layer.

That's could be a reason, but I didn't find from the paper how to mark the points, so I tried to randomly choose nodes for some minimal batch and train only on this batch to make a network to overfit this batch, but even in such conditions, it didn't converge.
